src/test/txpackage_tests.cpp

```diff
@@ -557,6 +577,182 @@ BOOST_FIXTURE_TEST_CASE(package_witness_swap_tests, TestChain100Setup)
+    Package package_still_too_low;
+    auto mtx_parent_cheap = CreateValidMempoolTransaction(/*input_transaction=*/ m_coinbase_txns[1], /*vout=*/ 0,
+                                                          /*input_height=*/ 0, /*input_signing_key=*/ coinbaseKey,
+                                                          /*output_destination=*/ parent_spk,
+                                                          /*output_amount=*/ coinbase_value, /*submit=*/ false);
+    CTransactionRef tx_parent_cheap = MakeTransactionRef(mtx_parent_cheap);
+    package_still_too_low.push_back(tx_parent_cheap);
+
+    auto mtx_child_cheap = CreateValidMempoolTransaction(/*input_transaction=*/ tx_parent_cheap, /*vout=*/ 0,
+                                                         /*input_height=*/ 101, /* input_signing_key */ child_key,
+                                                         /*output_destination=*/ child_spk,
+                                                         /*output_amount=*/ coinbase_value - 200, /*submit=*/ false);
+    CTransactionRef tx_child_cheap = MakeTransactionRef(mtx_child_cheap);
+    package_still_too_low.push_back(tx_child_cheap);
+
+    // Cheap package should fail with package-fee-too-low.
+    {
+        BOOST_CHECK_EQUAL(m_node.mempool->size(), expected_pool_size);
+        const auto submit_package_too_low = ProcessNewPackage(m_node.chainman->ActiveChainstate(), *m_node.mempool,
+                                                   package_still_too_low, /* test_accept */ false);
+        BOOST_CHECK_MESSAGE(submit_package_too_low.m_state.IsInvalid(), "Package validation unexpectedly succeeded");
+        BOOST_CHECK_EQUAL(submit_package_too_low.m_state.GetResult(), PackageValidationResult::PCKG_POLICY);
+        BOOST_CHECK_EQUAL(submit_package_too_low.m_state.GetRejectReason(), "package-fee-too-low");
+        BOOST_CHECK_EQUAL(m_node.mempool->size(), expected_pool_size);
+        const CFeeRate child_feerate(200, GetVirtualTransactionSize(*tx_child_cheap));
+        BOOST_CHECK(child_feerate.GetFeePerK() > 1000);
+        const CFeeRate expected_feerate(200,
+            GetVirtualTransactionSize(*tx_parent_cheap) + GetVirtualTransactionSize(*tx_child_cheap));
+        BOOST_CHECK(expected_feerate.GetFeePerK() < 1000);
+        BOOST_CHECK(submit_package_too_low.m_package_feerate != std::nullopt);
+        BOOST_CHECK_MESSAGE(submit_package_too_low.m_package_feerate.value() == expected_feerate,
+                            strprintf("Expected package feerate %s, got %s", expected_feerate.ToString(),
+                                      submit_package_too_low.m_package_feerate.value().ToString()));
+    }
+
+    // Package feerate includes the modified fees of the transactions.
+    // This means a child with its fee delta from prioritisetransaction can pay for a parent.
+    m_node.mempool->PrioritiseTransaction(tx_child_cheap->GetHash(), 1 * COIN);
```

Comment:
From checking `prioritisetransaction` documentation, I think it's a valid behavior to subtract fees from a transaction, thus altering `ProcessNewPackage` result. If you think it's a case worthy to cover.

Reply:
Good idea! I will add a test for this.